packages/sdk/test/ci/types/native-json.test.ts

```diff
@@ -0,0 +1,261 @@
+
+        // Test for numbers bigger than Number.MAX_SAFE_INTEGER
+        t = Timestamp.fromMillis(9007199254740997n);
+        e = '9007199254740997';
+        expect(JSONBig.stringify(t)).toEqual(e);
+    });
+});
+
+describe('Duration', () => {
+    test('Serializes as expected', () => {
+        let t = Duration.fromMillis(300);
+        let e = '300';
+
+        expect(JSONBig.stringify(t)).toEqual(e);
+        expect(() => JSON.stringify(t)).toThrowError(TypeError);
+        expect(testStringify(t)).toEqual(e);
+
+        // Test for numbers bigger than Number.MAX_SAFE_INTEGER
+        t = Duration.fromMillis(9007199254740997n);
+        e = '9007199254740997';
+        expect(JSONBig.stringify(t)).toEqual(e);
+    });
+});
+
+describe('ContractAddress', () => {
+    test('Serializes as expected', () => {
+        let t = ContractAddress.create(100, 10);
+        let e = '{"index":100,"subindex":10}';
+
+        expect(JSONBig.stringify(t)).toEqual(e);
+        expect(() => JSON.stringify(t)).toThrowError(TypeError);
+        expect(testStringify(t)).toEqual(e);
+
+        // Test for numbers bigger than Number.MAX_SAFE_INTEGER
+        t = ContractAddress.create(9007199254740997n, 10);
+        e = '{"index":9007199254740997,"subindex":10}';
+        expect(JSONBig.stringify(t)).toEqual(e);
+
+        t = ContractAddress.create(9007199254740997n, 109007199254740997n);
+        e = '{"index":9007199254740997,"subindex":109007199254740997}';
+        expect(JSONBig.stringify(t)).toEqual(e);
+    });
+});
+
+describe('Parameter', () => {
+    test('Serializes as expected', () => {
+        const t = Parameter.fromHexString('000102');
+        const e = '"000102"';
+
+        expect(JSONBig.stringify(t)).toEqual(e);
+    });
+});
+
+describe('TransactionHash', () => {
+    test('Serializes as expected', () => {
+        const v =
+            '1a17008f7944a5fd11a665a864266fb2d76794e754986c367455a4937fd3a66b';
+        const t = TransactionHash.fromHexString(v);
+        const e = `"${v}"`;
+
+        expect(JSONBig.stringify(t)).toEqual(e);
+    });
+});
+
+describe('BlockHash', () => {
+    test('Serializes as expected', () => {
+        const v =
+            '1a17008f7944a5fd11a665a864266fb2d76794e754986c367455a4937fd3a66b';
+        const t = BlockHash.fromHexString(v);
+        const e = `"${v}"`;
+
+        expect(JSONBig.stringify(t)).toEqual(e);
+    });
+});
+
+describe('ReturnValue', () => {
+    test('Serializes as expected', () => {
+        const t = ReturnValue.fromHexString('000102');
+        const e = '"000102"';
+
+        expect(JSONBig.stringify(t)).toEqual(e);
+    });
+});
+
+describe('ModuleReference', () => {
+    test('Serializes as expected', () => {
+        const v =
+            '5d99b6dfa7ba9dc0cac8626754985500d51d6d06829210748b3fd24fa30cde4a';
+        const t = ModuleReference.fromHexString(v);
+        const e = `"00000020${v}"`; // is prefixed with 4 byte length, for some reason
```

Comment:
```suggestion
        const e = `"00000020${v}"`; // is prefixed with 4 byte length
```
I don't think a comment on a test is the place to question the length of the length :confused:

Reply:
Well... I think you're right 😄